tasks/main.yml

```diff
@@ -1,38 +1,71 @@
 ---
-  - name: Installing SSSD and Kerberos dependencies using yum
+  - name: Installing Realmd, SSSD, and Kerberos dependencies using yum
     yum:
-        name: {{ linux_adbinding__dependencies_yum }}
-        state: present
-    when: ansible_pkg_mgr == yum
+      name: "{{ linux_adbinding__dependencies_yum }}"
+      state: present
+    when: ansible_pkg_mgr == "yum"
 
-  - name: Installing SSSD and Kerberos dependencies using apt
-    yum:
-        name: {{ linux_adbinding__dependencies_apt }}
-        state: present
-    when: ansible_pkg_mgr == apt
+  - name: Installing Realmd, SSSD, and Kerberos dependencies using apt
+    apt:
+      name: "{{ linux_adbinding__dependencies_apt }}"
+      state: present
+    when: ansible_pkg_mgr == "apt"
+
+  - name: Run Realm discover if not specified yet
+    shell: /bin/bash -c "/usr/sbin/realm discover | grep "domain-name" | head -n 1 | sed 's/.*:\s*//'"
+    register: realm_discover_output
+    changed_when: false
+    when: linux_adbinding__domain is undefined or linux_adbinding__domain == ''
+
+  - name: set fact for discovered variable
+    set_fact:
+      linux_adbinding__domain: "{{ realm_discover_output.stdout }}"
+    when: linux_adbinding__domain is undefined or linux_adbinding__domain == ''
+
+  - debug:
+      msg: "Specified or discovered domain is: {{ linux_adbinding__domain }}"
```

Comment:
```suggestion
      msg: "Specified or discovered domain is: {{ linux_adbinding__domain }}"
      verbosity: 2
```

Reply:
This only prints the variable if we are running in verbose mode.
Doc:
https://docs.ansible.com/ansible/latest/modules/debug_module.html